src/Vehicle/Vehicle.cc

```diff
@@ -263,6 +266,9 @@ void Vehicle::_sendMessage(mavlink_message_t message)
             static uint8_t messageKeys[256] = MAVLINK_MESSAGE_CRCS;
             mavlink_finalize_message_chan(&message, mavlink->getSystemId(), mavlink->getComponentId(), link->getMavlinkChannel(), message.len, messageKeys[message.msgid]);
             
+            // Give the plugin a chance to adjust
+            _firmwarePlugin->adjustMavlinkMessage(&message);
+            
```

Comment:
But whole if body looks little strange.

1. get buffer
2. finalize
3. adjust
4. write

I think that it should be:

1. adjust
2. finalize (if really needed)
3. get buffer
4. write